core/src/main/java/com/datastax/oss/driver/internal/core/config/cloud/CloudConfigFactory.java

```diff
@@ -228,8 +229,18 @@ protected BufferedReader fetchProxyMetadata(
       connection.setSSLSocketFactory(sslContext.getSocketFactory());
       connection.setRequestMethod("GET");
       connection.setRequestProperty("host", "localhost");
-      return new BufferedReader(
-          new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+      try {
+        return new BufferedReader(
+            new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+      } catch (FileNotFoundException e) {
+        throw new IllegalStateException(
+            "Metadata service request path not found. Please make sure the metadata service url is correct",
+            e);
```

Comment:
This change relies on undocumented behaviour (HttpsURLConnection throwing FNFE in this case).  We should either find a more conventional way to note this case or else remove the special logging for this specific code path.